imap_processing/codice/constants.py

```diff
@@ -406,19 +442,59 @@
     },
 }
 
-# Dictionary to define the bit structure of CoDICE-Lo direct events
-LO_DE_BIT_STRUCTURE = {
-    "APDGain": 1,
-    "APD_ID": 5,
-    "Position": 5,
-    "APDEnergy": 9,
-    "TOF": 10,
-    "MultiFlag": 1,
-    "PHAType": 2,
-    "SpinAngle": 5,
-    "EnergyStep": 7,
-    "Priority": 3,
-    "Spare": 16,
+# Various configurations to support processing of direct events data products
+# These are described in the algorithm document in chapter 10 ("Data Level 1A")
+DE_DATA_PRODUCT_CONFIGURATIONS: dict[Any, dict[str, Any]] = {
+    CODICEAPID.COD_HI_PHA: {
+        "num_priorities": 6,
+        "bit_structure": {
+            "SSDEnergy": {"bit_length": 11, "dtype": np.uint16, "fillval": 65535},
```

Comment:
If you don't want to specify all the fillvals and they are all the maximums you could put something in the code above like:
```python
fill_val = np.iinfo(dtype_field).max
```